### PR TITLE
CORE-9167 - Persist mgm info on member side

### PR DESCRIPTION
**Pull request by nikinagy**
During virtual node creation and upgrade, we parse the MGM information from the group policy file in the CPI and publish it to kafka. This means that the information only exists in the member list on kafka, or in the group policy file.
For other members (i.e. everyone other than the MGM) the member list will be received from the MGM, persisted to the database, and published to kafka. 
In the case of kafka failure, the virtual node database reconciliation will help recovery by reconciling the member list from the database to kafka. As it currently stands, the mgm info would need to be retrieved from the group policy, adding additional complexity to the reconciliation which should be as lightweight as possible. 
To simplify the member list handling and the reconciliation, we should persist the MGMs member info to the member’s member list database table when we also publish to kafka so that the reconciliation can recover easier in case of kafka failure.

**Solution**
Because the VirtualNodeInfoProcessor could not process the vnode record required to access the vnode vault DB in time, I decided to move the mGM info publishing and persistence to the dynamic registration service instead.

**Testing**
After successful registration MGM information got persisted into member DB:
Screenshot 2024-04-29 at 12 38 49

E2E tests build: https://ci02.dev.r3.com/blue/organizations/jenkins/Corda5%2Fcorda5-e2e-tests/detail/release%2F5.2/1137/pipeline
Multi cluster E2E build: https://ci02.dev.r3.com/blue/organizations/jenkins/Corda5%2Fcorda-e2e-tests-multi-cluster-tests/detail/nikinagy%2Ftest/12/pipeline

### CODE REVIEW

File: components/virtual-node/virtual-node-write-service-impl/src/main/kotlin/net/corda/virtualnode/write/db/impl/writer/asyncoperation/handlers/CreateVirtualNodeOperationHandler.kt [hunk 1]
```diff
@@ -25,7 +24,6 @@ internal class CreateVirtualNodeOperationHandler(
     private val createVirtualNodeService: CreateVirtualNodeService,
     private val virtualNodeDbFactory: VirtualNodeDbFactory,
     private val recordFactory: RecordFactory,
-    private val policyParser: GroupPolicyParser,
```
Comment (vkolomeyko) [inline components/virtual-node/virtual-node-write-service-impl/src/main/kotlin/net/corda/virtualnode/write/db/impl/writer/asyncoperation/handlers/CreateVirtualNodeOperationHandler.kt:28]: Should we try to remove dependency from this module to `:libs:membership:membership-common` ?

File: components/virtual-node/virtual-node-write-service-impl/src/main/kotlin/net/corda/virtualnode/write/db/impl/writer/VirtualNodeWriterFactory.kt [hunk 2]
```diff
@@ -61,8 +59,6 @@ internal class VirtualNodeWriterFactory(
     private val dbConnectionManager: DbConnectionManager,
     private val virtualNodeDbAdmin: VirtualNodesDbAdmin,
     private val schemaMigrator: LiquibaseSchemaMigrator,
-    private val groupPolicyParser: GroupPolicyParser,
-    private val membershipGroupReaderProvider: MembershipGroupReaderProvider,
```
Comment (vkolomeyko) [inline components/virtual-node/virtual-node-write-service-impl/src/main/kotlin/net/corda/virtualnode/write/db/impl/writer/VirtualNodeWriterFactory.kt:65]: Should we try to remove dependency from this module on `:components:membership:membership-group-read` ?

**Review by vkolomeyko — COMMENTED** [review]
Looks fine from @corda/rest point of view. Few minor comments posted.

**Comment by vkolomeyko** [comment]
build e2e

File: components/virtual-node/virtual-node-write-service-impl/src/main/kotlin/net/corda/virtualnode/write/db/impl/writer/asyncoperation/handlers/CreateVirtualNodeOperationHandler.kt (hunk 1 above)
Comment (nikinagy) [reply components/virtual-node/virtual-node-write-service-impl/src/main/kotlin/net/corda/virtualnode/write/db/impl/writer/asyncoperation/handlers/CreateVirtualNodeOperationHandler.kt:28]: addressed in [7f74a9b](https://github.com/corda/corda-runtime-os/pull/6082/commits/7f74a9bfbba41fc7437b4878fdcd2c651f36778f)

File: components/virtual-node/virtual-node-write-service-impl/src/main/kotlin/net/corda/virtualnode/write/db/impl/writer/VirtualNodeWriterFactory.kt (hunk 2 above)
Comment (nikinagy) [reply components/virtual-node/virtual-node-write-service-impl/src/main/kotlin/net/corda/virtualnode/write/db/impl/writer/VirtualNodeWriterFactory.kt:65]: addressed in [7f74a9b](https://github.com/corda/corda-runtime-os/pull/6082/commits/7f74a9bfbba41fc7437b4878fdcd2c651f36778f)

**Review by vkolomeyko — APPROVED** [review]
LGTM from @corda/rest point of view

**Review by YashNabar — APPROVED** [review]
LGTM

**Pull request merged**